pkg/ebpf/uprobes/attacher.go

```diff
@@ -0,0 +1,901 @@
+
+	if ua.config.PerformInitialScan {
+		// Initial scan only looks at existing processes, and as it's the first scan
+		// we don't have to track deletions
+		err := ua.Sync(true, false)
+		if err != nil {
+			return fmt.Errorf("error during initial scan: %w", err)
+		}
+	}
+
+	if ua.handlesLibraries() {
+		if !sharedlibraries.IsSupported(ua.config.EbpfConfig) {
+			return errors.New("shared libraries tracing not supported for this platform")
+		}
+
+		ua.soWatcher = sharedlibraries.NewEBPFProgram(ua.config.EbpfConfig)
+
+		err := ua.soWatcher.Init()
+		if err != nil {
+			return fmt.Errorf("error initializing shared library program: %w", err)
+		}
+		err = ua.soWatcher.Start()
+		if err != nil {
+			return fmt.Errorf("error starting shared library program: %w", err)
+		}
+	}
+
+	ua.wg.Add(1)
+	go func() {
+		processSync := time.NewTicker(ua.config.ScanProcessesInterval)
+
+		defer func() {
+			processSync.Stop()
+			if cleanupExec != nil {
+				cleanupExec()
+			}
+			cleanupExit()
+			procMonitor.Stop()
+			ua.fileRegistry.Clear()
+			if ua.soWatcher != nil {
+				ua.soWatcher.Stop()
+			}
+			ua.wg.Done()
+			log.Infof("uprobe attacher %s stopped", ua.name)
+		}()
+
+		var sharedLibDataChan <-chan *ebpf.DataEvent
+		var sharedLibLostChan <-chan uint64
+
+		if ua.soWatcher != nil {
+			sharedLibDataChan = ua.soWatcher.GetPerfHandler().DataChannel()
+			sharedLibLostChan = ua.soWatcher.GetPerfHandler().LostChannel()
+		}
+
+		for {
+			select {
+			case <-ua.done:
+				return
+			case <-processSync.C:
+				// We always track process deletions in the scan, to avoid memory leaks.
+				_ = ua.Sync(ua.config.EnablePeriodicScanNewProcesses, true)
+			case event, ok := <-sharedLibDataChan:
+				if !ok {
+					return
+				}
+				_ = ua.handleLibraryOpen(event)
+			case <-sharedLibLostChan:
+				// Nothing to do in this case
+				break
+			}
+		}
+	}()
+	log.Infof("uprobe attacher %s started", ua.name)
+
+	return nil
+}
+
+// Sync scans the proc filesystem for new processes and detaches from terminated ones
+func (ua *UprobeAttacher) Sync(trackCreations bool, trackDeletions bool) error {
+	if !trackDeletions && !trackCreations {
+		return nil // Nothing to do
+	}
+
+	var deletionCandidates map[uint32]struct{}
+	if trackDeletions {
+		deletionCandidates = ua.fileRegistry.GetRegisteredProcesses()
+	}
+	thisPID, err := kernel.RootNSPID()
+	if err != nil {
+		return err
+	}
+
+	_ = kernel.WithAllProcs(ua.config.ProcRoot, func(pid int) error {
+		if pid == thisPID { // don't scan ourselves
+			return nil
+		}
+
+		if trackDeletions {
+			if _, ok := deletionCandidates[uint32(pid)]; ok {
+				// We have previously hooked into this process and it remains active,
+				// so we remove it from the deletionCandidates list, and move on to the next PID
+				delete(deletionCandidates, uint32(pid))
+				return nil
+			}
+		}
+
+		if trackCreations {
+			// This is a new PID so we attempt to attach SSL probes to it
+			_ = ua.AttachPID(uint32(pid))
+		}
```

Comment:
Continuing https://github.com/DataDog/datadog-agent/pull/27663/files#r1745282258

Having both of them set together seems like a weird behavior